### DISCUSSION TITLE
V1 RunTime Announcement

### DISCUSSION THREAD

**Discussion by Mister-Hope** [vuepress-theme-hope/vuepress-theme-hope#1099] [Annoucement]
We are planning to bundle a runtime together with theme in future versions of VuePress Theme Hope V1.

## Reasons

There are several reasons of us to do it. The most basic one is that Evan You marked VitePress as the future version of VuePress. This results:

- VuePress V1 is lacking maintenance for years, and it it's using a very old tool chain
- The future version of VuePress (v2) is more likely meteorlxy's personal project. He is the only maintainer now, so there won't be any Vue core team behind the scene.

We are excited that VuePress V2 decoupled bundler, so that users can use either Webpack or Vite to bundle their project. But VuePress@V2 is targeting Vue3, and there may be some people still staying on Vue2. Also, VuePress change it's internal structure to improve scalability, so that you must access other page data asynchronously. So that the blog feature will be diffcult to implement. (The official still not support yet, and there will be huge changes with it, probably will still need to wait)

Under these backgrounds, for these users:

- People who want to continue using Vue2
- People who want to migrate to webpack 5 or use other newer versions of deps, but getting deps-related issue
- People who are using VuePress V1 as a blog builder

We provide a forked version of VuePress V1 to you as a better runtime targeting V1.

## Changes

We will release several packages, including:

- `@mr-hope/vuepress-core`: Improved version of `@vuepress/core`, coming with several plugins and loaders update, also with postcss8 and webpack5(Maybe, but not sure because `webpack-chain` does not work on V5 so that `chainWebpack` won't work after the change)
- `@mr-hope/vuepress-markdown`: Improved version of `@vuepress/markdown`, coming with MarkdownIt12, and with serveral bug fixes. MarkdownIt 12 has better performance.
- `@mr-hope/vuepress-markdown-loader`: Improved version of `@vuepress/markdown-loader`
- `@mr-hope/vuepress-shared-utils`: Improved version of `@vuepress/shared-utils`
- `@mr-hope/vuepress-test-utils`: Improved version of `@vuepress/test-utils` with Jest 27
- `@mr-hope/vuepress-runtime`: Cli of this runtime, replacing `VuePress`

### Dropping non-LTS Node Versions

This runtime will only support LTS versions of Node, which means we are dropping Node 8 and Node 10, and starting support with Node 12.12. So that we can update our deps and increase performance.

### Cli command changes

Since this runtime will be bundled in future versions of VuePress Theme Hope V1, we will use `hopepress` to replace `vuepress` comand.

### Upgrading to Webpack5 (Not sure)

Meeting diffculty upgrading to Webpack5 to make SSR work, VuePress V1 internal loaders needs a full rewrite to work, so this will be shipped once completed.

**Comment by AndersonHJB** [comment]
希望有到时候有 vuepress1 或者之类的教程，非常期待！谢谢

**Comment by Mister-Hope** [comment]
Due to uliuz is back patching vuepress and fix security problems, runtime will not be published.
